### PR TITLE
Changed info overlay to use an interface and new info overlay variables

### PR DESCRIPTION
**Pull request by aws404**
This rewrites portions of the InfoOverlay. 
InfoOverlay variables are now defined in the constructer of the InfoFormatter class in the form of the InfoModule interface assigned into a HashMap. This means when a string is found between to % signs, an attempt to get that key from the HashMap is made and if successful, the output is generated.
Some InfoModules use the cache HashMap, replacing the variables that were being used previously.

**New Info Variables:**
- %kmph% - Kilometers per hour (1000 blocks per hour)
- %mana% - Current mana
- %mana_max% - Max mana
- %health% - Current health
- %health_max% - Max health
- %xp% - Current XP (Formatted)
- %xp_raw% - Current XP (Raw)
- %xp_req% - Required XP to level up (Formatted)
- %xp_pct% - Percentage XP to next level
- %horse_xp% - Current horse XP (Bascially percentage)
- %horse_level% - Current horse level (Jump/Speed level)
- %horse_level_max% - Max horse level
- %horse_tier% - The horse tier (ie. Brown Horse = 1, Black Horse = 2, ect)
- %pouch% - Amount of items in the ingredeint pouch
- %sp_timer% / %soulpoint_timer% - A simple timer until next soul point (Format: mm:ss)
- %sp_timer_s% / %soulpoint_timer_s% - Seconds until next soul point (in relation to minutes)
- %sp_timer_m% / %soulpoint_timer_m% - Minutes until next soul point (in relation to seconds)
- %sp% / %soulpoints% - current soul points
- %sp_max% / %soulpoints_max% - max soul points
- %potions_health% - number of health potions in the inventory
- %potions_mana% - number of mana potions in the inventory
- %party_count% - number of players in the party
- %party_owner% - the party owner

**New Info Symbols**
- \M - mana symbol
- \H - health/heart symbol

**New Settings**
- Text alignment options (per overlay)
- Custom background colour (global)
- Variable list (cycles through each variable and copies it to the clipboard, same as presets)
- Added to presets list

**Misc Changes**
- The rendering was updated to use updated methods and more options.
- The morse mount keybind now attempts to find a saddle and spawn a horse if one is found. (now possible due to horse tracking)

### CODE REVIEW

**Review by magicus — APPROVED** [review]
Nice!

File: src/main/java/com/wynntils/core/framework/instances/PlayerInfo.java [hunk 1]
```diff
@@ -253,6 +256,42 @@ public int getMoney() {
...
+     * @return Total number of health potions in inventory
+     */
+    public int getHealthPotions() {
+        if (mc.player == null) return 0;
+        NonNullList<ItemStack> contents = mc.player.inventory.mainInventory;
+        
+        int count = 0;
+        
+        for (ItemStack item : contents) {
+            if (item.hasDisplayName() && item.getDisplayName().contains("Potion of Healing")) {
```
Comment (coehlrich) [inline src/main/java/com/wynntils/core/framework/instances/PlayerInfo.java]: Should have a check for `item.isEmpty()`

File: src/main/java/com/wynntils/core/framework/instances/PlayerInfo.java [hunk 2]
```diff
@@ -253,6 +256,42 @@ public int getMoney() {
...
+     * @return Total number of mana potions in inventory
+     */
+    public int getManaPotions() {
+        if (mc.player == null) return 0;
+        NonNullList<ItemStack> contents = mc.player.inventory.mainInventory;
+        
+        int count = 0;
+        
+        for (ItemStack item : contents) {
+            if (item.hasDisplayName() && item.getDisplayName().contains("Potion of Mana")) {
```
Comment (coehlrich) [inline src/main/java/com/wynntils/core/framework/instances/PlayerInfo.java]: Should have a check for `item.isEmpty()`

File: src/main/java/com/wynntils/core/framework/instances/PlayerInfo.java [hunk 3]
```diff
@@ -337,5 +376,89 @@ public int getTicksToNextSoulPoint() {
...
+     */
+    public HorseData getHorseData() {
+        if (currentClass == ClassType.NONE || mc.player == null) return null;
+        
+        NonNullList<ItemStack> inventory = mc.player.inventory.mainInventory;
+        
+        if (horseData != null) {
+            ItemStack stack = inventory.get(horseData.inventorySlot);
+            
+            if (stack.hasDisplayName() && stack.getDisplayName().contains(" Horse")) {
```
Comment (coehlrich) [inline src/main/java/com/wynntils/core/framework/instances/PlayerInfo.java]: Should have a check for `stack.isEmpty()`

File: src/main/java/com/wynntils/core/framework/instances/PlayerInfo.java [hunk 4]
```diff
@@ -337,5 +376,89 @@ public int getTicksToNextSoulPoint() {
...
+            if (stack.hasDisplayName() && stack.getDisplayName().contains(" Horse")) {                
+                horseData = new HorseData(stack, horseData.inventorySlot);
+                return horseData;
+            }
+        }
+        
+        for (int i = 0; i < inventory.size(); i++) {
+            ItemStack stack = inventory.get(i);
+            
+            if (stack.hasDisplayName() && stack.getDisplayName().contains(" Horse")) {
```
Comment (coehlrich) [inline src/main/java/com/wynntils/core/framework/instances/PlayerInfo.java]: Should have a check for `stack.isEmpty()`

File: src/main/java/com/wynntils/core/utils/StringUtils.java [hunk 5]
```diff
@@ -341,5 +341,16 @@ public static boolean isWynnicNumber(char character) {
         return (0x2474 <= character && character <= 0x247C)
             || (0x247D <= character && character <= 0x247F);
     }
+    /**
+     * @param count the number
+     * @return The formatted shortened string
+     */
+    public static String integerToShortString(int count) {
+        if (count < 1000) return "" + count;
```
Comment (coehlrich) [inline src/main/java/com/wynntils/core/utils/StringUtils.java]: Use `Integer.toString`

File: src/main/java/com/wynntils/modules/utilities/overlays/hud/InfoOverlay.java [hunk 6]
```diff
@@ -5,299 +5,106 @@
...
+                // Draw Text
+                for (int i = 0; i < lines.length; i++) {
+                    drawString(lines[i], textOrigin, i*11, CommonColors.WHITE, getAlignment(), OverlayConfig.InfoOverlays.INSTANCE.textShadow);
+                }
             }
         }
-
     }
-
+
```
Comment (coehlrich) [inline src/main/java/com/wynntils/modules/utilities/overlays/hud/InfoOverlay.java]: Remove this tab

File: src/main/java/com/wynntils/modules/utilities/managers/MountHorseManager.java [hunk 7]
```diff
@@ -60,7 +62,19 @@ public static MountHorseStatus mountHorse() {
...
+                return MountHorseStatus.NO_HORSE;
+            }
+            
+            int prev = mc.player.inventory.currentItem;
+            
+            mc.player.inventory.currentItem = horse.inventorySlot;
+            mc.playerController.processRightClick(player, player.world, EnumHand.MAIN_HAND);
+            mc.player.inventory.currentItem = prev;
+            
+            return mountHorse(false);
```
Comment (coehlrich) [inline src/main/java/com/wynntils/modules/utilities/managers/MountHorseManager.java]: The horse would need to spawn before the player can mount it

**Review by coehlrich — CHANGES_REQUESTED** [review]

File: src/main/java/com/wynntils/modules/utilities/managers/MountHorseManager.java (hunk 7 above)
Comment (aws404) [reply src/main/java/com/wynntils/modules/utilities/managers/MountHorseManager.java]: I changed the way this works to activate the mount when a new horse is spawned, it appears to be more stable this way. Is this a good solution?

**Review by coehlrich — APPROVED** [review]

**Pull request merged**